src/profile-logic/profile-data.js

```diff
@@ -2784,15 +2784,24 @@ export function extractProfileFilterPageData(
     return null;
   }
 
-  // Getting the first page's innerWindowID and then getting its url.
-  const innerWindowID = [...relevantPages][0];
-  const filteredPages = pages.filter(
-    page => page.innerWindowID === innerWindowID
+  // Getting the pages that are relevant and a top-most frame.
+  let filteredPages = pages.filter(
+    page =>
+      // It's the top-most frame if `embedderInnerWindowID` is zero.
+      relevantPages.has(page.innerWindowID) && page.embedderInnerWindowID === 0
```

Comment:
super nit: I'd put `page.embedderInnerWindowID === 0` first because it's a fast check, it will short-circuit if it's false.